src/Orchard.Web/Modules/Orchard.Comments/Drivers/CommentsPartDriver.cs

```diff
@@ -77,6 +77,22 @@ public class CommentsPartDriver : ContentPartDriver<CommentsPart> {
                         return shapeHelper.Parts_Comments_Count(
                             CommentCount: part.CommentsCount);
                     }),
+                ContentShape("Parts_Comments_Count_Detail",
+                    () => {
+                        if (part.CommentsShown == false)
+                            return null;
+
+                        return shapeHelper.Parts_Comments_Count_Detail(
+                            CommentCount: part.CommentsCount);
+                    }),

```

Comment:
You are right. I put it only because "sounds" coherent. But I'm sure we can avoid it... sorry for the extra block